runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -1355,43 +1386,70 @@ StoreArrayElementTransformer::lower(TR::Node* const node, TR::TreeTop* const tt)
    TR::SymbolReference *elementSymRef = comp->getSymRefTab()->findOrCreateArrayShadowSymbolRef(TR::Address, anchoredArrayBaseAddressNode);
    TR::Node *elementStoreNode = TR::Node::createWithSymRef(TR::awrtbari, 3, 3, elementAddress, anchoredValueNode, anchoredArrayBaseAddressNode, elementSymRef);
 
-   TR::SymbolReference *arrayStoreCHKSymRef = comp->getSymRefTab()->findOrCreateTypeCheckArrayStoreSymbolRef(comp->getMethodSymbol());
-   TR::Node *arrayStoreCHKNode = TR::Node::createWithRoomForThree(TR::ArrayStoreCHK, elementStoreNode, 0, arrayStoreCHKSymRef);
-   arrayStoreCHKNode->copyByteCodeInfo(node);
-
-   TR::TreeTop *arrayStoreCHKTT = originalBlock->append(TR::TreeTop::create(comp, arrayStoreCHKNode));
-
-   if (enableTrace)
-      traceMsg(comp, "Created arrayStoreCHK treetop n%dn node n%dn\n", arrayStoreCHKTT->getNode()->getGlobalIndex(), arrayStoreCHKNode->getGlobalIndex());
+   TR::Node *arrayStoreCHKNode = NULL;
+   TR::TreeTop *arrayStoreTT = NULL;
+   bool addArrayStoreCheck = !skipArrayStoreChecks(comp, node);
 
+   // Some JCL methods are known never to trigger an ArrayStoreException
+   // Only add an ArrayStoreCHK if it's required
+   if (addArrayStoreCheck)
```

Comment:
I realized addArrayStoreCheck could be eliminated
```c++
if (!skipArrayStoreChecks(comp, node))
   {
   ...
   }
else 
   {
   ... 
   }
...
   if (enableTrace)
      traceMsg(comp, "Isolated array element store treetop n%dn node n%dn in block_%d\n", arrayStoreTT->getNode()->getGlobalIndex(),
            arrayStoreCHKNode ? arrayStoreCHKNode->getGlobalIndex() : elementStoreNode->getGlobalIndex(), arrayElementStoreBlock->getNumber());
```

Reply:
`addArrayStoreCheck` is also used in some trace code below, so it's still needed.

Reply:
It's only used once in the trace. The above snippet of code suggests checking `arrayStoreCHKNode` instead of checking `addArrayStoreCheck`

` arrayStoreCHKNode ? arrayStoreCHKNode->getGlobalIndex() : elementStoreNode->getGlobalIndex()`

Reply:
Sorry - I missed that.  I can make that change.

Reply:
Fixed in commit 589ca976ec12bda5b9b10fc5a15589f722b060e7